### PR TITLE
Added PWA Support

### LINKED ISSUE #633: Add PWA support
Workaround for #429.

### PR DESCRIPTION
**Pull request by johnmgrant**
Closes #633

* Adds PoC for PWA support as a compromise for #429 on older iOS devices
  * PWA support confirmed to alleviate team analysis lag issue

### CODE REVIEW

**Comment by RaenonX** [comment]
@johnmgrant See if changes I made could cause some issue to PWA? Mainly moving where metadata is located and update some properties accordingly.

`viewport` is deprecated so I just removed it from the PWA metadata.

**Comment by johnmgrant** [comment]
That worked better than before! I was planning on fixing the errors if you were in favor of the idea, but it's no longer throwing any errors and the theme colors and icons are appearing better as well.

https://github.com/RaenonX-PokemonSleep/pokemon-sleep-ui/assets/10672193/852a39e5-4b30-4465-8d9f-21da148f9160

**Comment by RaenonX** [comment]
> That worked better than before! I was planning on fixing the errors if you were in favor of the idea, but it's no longer throwing any errors and the theme colors and icons are appearing better as well.
> 
>  pksbyraenonx_pwa.mov

Nice! Could you run a `yarn install` on your local, and if CI passes, change this to a formal PR?

File: .gitignore [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
-# vercel
+# Vercel
 .vercel
 
-# typescript
+# Typescript
 *.tsbuildinfo
 next-env.d.ts
 
 # logs
```
Comment (RaenonX) [inline .gitignore:39]: Capitalize

**Review by RaenonX — CHANGES_REQUESTED** [review]
For the 2nd commit, `<RootLayout/>` already calls it, doesn't that work? Technically that kinda of site-wide setup shouldn't require copy-pasta in every page, so if that's needed, then there should be a wrapper component that calls it and everyone uses it (which is `<RootLayout/>` in this case.

**Comment by johnmgrant** [comment]
> For the 2nd commit, `<RootLayout/>` already calls it, doesn't that work? Technically that kinda of site-wide setup shouldn't require copy-pasta in every page, so if that's needed, then there should be a wrapper component that calls it and everyone uses it (which is `<RootLayout/>` in this case.

I've reverted most of this copy/ paste change and left the only one needed to resolve the error seen in the PWA.

**Comment by RaenonX** [comment]
> > For the 2nd commit, `<RootLayout/>` already calls it, doesn't that work? Technically that kinda of site-wide setup shouldn't require copy-pasta in every page, so if that's needed, then there should be a wrapper component that calls it and everyone uses it (which is `<RootLayout/>` in this case.
> 
> I've reverted most of this copy/ paste change and left the only one needed to resolve the error seen in the PWA.

I see both `<RootLayout/>` and `<PageLayout/>` calling `unstable_setRequestLocale()`. Doesn't calling it in `<RootLayout/>` work?

Also maybe squash the 2nd commit into 1st once finalized because they're all PWA support-related and it's not 20+ file change

**Comment by johnmgrant** [comment]
> > > For the 2nd commit, `<RootLayout/>` already calls it, doesn't that work? Technically that kinda of site-wide setup shouldn't require copy-pasta in every page, so if that's needed, then there should be a wrapper component that calls it and everyone uses it (which is `<RootLayout/>` in this case.
> 
> > 
> 
> > I've reverted most of this copy/ paste change and left the only one needed to resolve the error seen in the PWA.
> 
> 
> 
> I see both `<RootLayout/>` and `<PageLayout/>` calling `unstable_setRequestLocale()`. Doesn't calling it in `<RootLayout/>` work?
> 
> 
> 
> Also maybe squash the 2nd commit into 1st once finalized because they're all PWA support-related and it's not 20+ file change

I don't think it was because I was seeing the error thrown without the change here. I found the documentation for i18n in next.js [docs](https://next-intl-docs.vercel.app/docs/getting-started/app-router#add-unstable_setrequestlocale-to-all-layouts-and-pages) and saw that it specifically says to add this to “…all pages and layouts where static rendering is required since Next.js can render layouts and pages independently.”

+1 to squashing this change now though. I'll do that when I'm back by my computer later today.

**Comment by RaenonX** [comment]
> > > > For the 2nd commit, `<RootLayout/>` already calls it, doesn't that work? Technically that kinda of site-wide setup shouldn't require copy-pasta in every page, so if that's needed, then there should be a wrapper component that calls it and everyone uses it (which is `<RootLayout/>` in this case.
> > 
> > 
> > > 
> > 
> > 
> > > I've reverted most of this copy/ paste change and left the only one needed to resolve the error seen in the PWA.
> > 
> > 
> > I see both `<RootLayout/>` and `<PageLayout/>` calling `unstable_setRequestLocale()`. Doesn't calling it in `<RootLayout/>` work?
> > Also maybe squash the 2nd commit into 1st once finalized because they're all PWA support-related and it's not 20+ file change
> 
> I don't think it was enough since I was seeing the error thrown without the change I added. I found the documentation for i18n in next.js [docs](https://next-intl-docs.vercel.app/docs/getting-started/app-router#add-unstable_setrequestlocale-to-all-layouts-and-pages) and saw that it specifically says to add this to “…all pages and layouts where static rendering is required since Next.js can render layouts and pages independently.”
> 
> +1 to squashing this change now though. I'll do that when I'm back by my computer later today.

Got it. What if you remove the call in `<RootLayout/>` and keep it in `<PageLayout/>` only? I think this `<PageLayout/>` is called in every page.

**Comment by johnmgrant** [comment]
> Got it. What if you remove the call in `<RootLayout/>` and keep it in `<PageLayout/>` only? I think this `<PageLayout/>` is called in every page.

No issue seen after making this change. 👍

**Review by RaenonX — APPROVED** [review]

**Comment by RaenonX** [comment]
> > Got it. What if you remove the call in `<RootLayout/>` and keep it in `<PageLayout/>` only? I think this `<PageLayout/>` is called in every page.
> 
> No issue seen after making this change. 👍

Thanks for your work! Did a minor touch about statement grouping ~~because of my OCD~~.

**Pull request merged**